### PR TITLE
IT-10656 Handling situation when 'results' key is missing in response

### PR DESCRIPTION
**Pull request by gecbla**

### CODE REVIEW

**Review by nemke — DISMISSED** [review]

**Review by AnthonyPorthouse — COMMENTED** [review]
When we have an exception we shouldn't be setting the results key at all. The calling code should handle this by checking for the response status first.

File: src/Arbor/Api/Gateway/RestGateway.php [hunk 1]
```diff
@@ -302,6 +302,10 @@ public function bulkCreate(string $resource, Collection $collection, bool $check
         }
 
         if (isset($exception)) {
```
Comment (AnthonyPorthouse) [inline src/Arbor/Api/Gateway/RestGateway.php]: This entire block should be refactored into the above catch block to remove the need for the `if (isset($exception)) {` block.

**Review by AnthonyPorthouse — CHANGES_REQUESTED** [review]

File: src/Arbor/Api/Gateway/RestGateway.php (hunk 1 above)
Comment (gecbla) [reply src/Arbor/Api/Gateway/RestGateway.php]: @AnthonyPorthouse This part is refacoted now, so can you check PR now

**Review by davorzdralo — APPROVED** [review]

**Review by aaronlinley — APPROVED** [review]

**Comment by gecbla** [comment]
@AnthonyPorthouse Can I get a ✅  as you were the one requesting changes?

**Review by AnthonyPorthouse — APPROVED** [review]

**Pull request merged**